x/nft/keeper/grpc_query.go

```diff
@@ -65,58 +65,68 @@ func (k Keeper) Supply(goCtx context.Context, r *nft.QuerySupplyRequest) (*nft.Q
 	return &nft.QuerySupplyResponse{Amount: supply}, nil
 }
 
-// NFTsOfClass return all NFTs of a given class or optional owner, similar to tokenByIndex in ERC721Enumerable
-func (k Keeper) NFTsOfClass(goCtx context.Context, r *nft.QueryNFTsOfClassRequest) (*nft.QueryNFTsOfClassResponse, error) {
+// NFTs queries all NFTs of a given class or owner (at least one must be provided), similar to tokenByIndex in ERC721Enumerable
```

Comment:
I just checked https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-043-nft-module.md and there's one hidden assumption that a class id isn't empty.
I assume that would be the case everywhere, but perhaps there may be a network that has a custom NFT module where all tokens fall under one class with an empty id or something like that.
It'll be good to add this assumption in a comment + in ADR-043

Reply:
> looks all right, but it'll be good to document the class id format (length, regex, ...) assumptions more explicitly

Ok, in adr-043, I added the regular check for class_id and nft_id

Reply:
> I just checked https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-043-nft-module.md and there's one hidden assumption that a class id isn't empty. I assume that would be the case everywhere, but perhaps there may be a network that has a custom NFT module where all tokens fall under one class with an empty id or something like that. It'll be good to add this assumption in a comment + in ADR-043
For the scene where class_id does not exist, I think it is still rare. If there is such a demand, I think it can also be solved by issuing a class, and then all nfts are mint under this class.

Reply:
> I just checked https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-043-nft-module.md and there's one hidden assumption that a class id isn't empty. I assume that would be the case everywhere, but perhaps there may be a network that has a custom NFT module where all tokens fall under one class with an empty id or something like that. It'll be good to add this assumption in a comment + in ADR-043

For the scene where class_id does not exist, I think it is still rare. If there is such a demand, I think it can also be solved by issuing a class, and then all nfts are mint under this class.